### PR TITLE
Add cluster claim for hypershift clusters

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>
issue reference: https://github.com/stolostron/backlog/issues/20056

Will use https://github.com/stolostron/backlog/issues/21241 to trace the permission problem.

### CODE REVIEW

File: pkg/agent/clusterclaim.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"k8s.io/apimachinery/pkg/types"
+
+	clusterclientset "open-cluster-management.io/api/client/cluster/clientset/versioned"
+
+	clusterv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+)
+
+const (
+	// labelExcludeBackup is true for the local-cluster will be not backed up into velero
+	labelExcludeBackup = "velero.io/exclude-from-backup"
```
Comment (zhujian7) [inline pkg/agent/clusterclaim.go:21]: @zhiweiyin318 Please help to make sure this is correct?

**Comment by zhujian7** [comment]
cc @ianzhang366 @qiujian16 @zhiweiyin318

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -305,7 +325,27 @@ func (c *agentController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
...
 
-	return ctrl.Result{}, createOrUpdateMirrorSecrets()
+	if err := createOrUpdateMirrorSecrets(); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := c.createHostedClusterClaim(ctx, types.NamespacedName{Namespace: hc.Namespace, Name: hc.Status.KubeConfig.Name},
+		generateClusterClientFromSecret); err != nil {
+		if strings.Contains(err.Error(), "the server could not find the requested resource") ||
+			strings.Contains(err.Error(), "no such host") ||
```
Comment (qiujian16) [inline pkg/agent/agent.go:335]: I am not sure if it is necessary to iterate all the error message. To keep it simple, maybe always requeue when there is an error

File: pkg/agent/clusterclaim.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+	hypershiftManagementClusterClaimKey = "managementcluster.hypershift.openshift.io"
+	hypershiftHostedClusterClaimKey     = "hostedcluster.hypershift.openshift.io"
+)
+
+func newClusterClaim(name, value string) *clusterv1alpha1.ClusterClaim {
+	return &clusterv1alpha1.ClusterClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   name,
+			Labels: map[string]string{labelExcludeBackup: "true"},
```
Comment (qiujian16) [inline pkg/agent/clusterclaim.go:31]: why this should be labled to backup?

Comment (zhujian7) [reply pkg/agent/clusterclaim.go:31]: I want it *exclude* from backup, @zhiweiyin318 do we need this here?

File: pkg/agent/clusterclaim.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/agent/clusterclaim.go:21]: I think it is correct.

File: pkg/agent/clusterclaim.go (hunk 3 above)
Comment (zhiweiyin318) [reply pkg/agent/clusterclaim.go:31]: yes. I think the label is to exclude resources from backup

**Comment by qiujian16** [comment]
it looks good to me

**Pull request merged**